students/mark_luckeroth/lesson04/mailroom2.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+import os
+
+donations = {
+    'Peter Pan': [10., 10., 10.],
+    'Paul Hollywood': [5., 50000., 5., 5.],
+    'Mary Berry': [100.],
+    'Jake': [123., 456., 789.],
+    'Raj': [60., 600000.]}
+
+main_prompt = ("\nWould you like to:\n"
+              "1 - Send a Thank You\n"
+              "2 - Create a Report\n"
+              "3 - Send letters to everyone\n"
+              "4 - Quit\n")
+
+def totals(donations):
+    totals = {}
+    for item in donations:
+        totals[item] = sum(donations[item])
+    return totals
+
+def count(donations):
+    count = {}
+    for item in donations:
+        count[item] = len(donations[item])
+    return count
+
+def averages(d_totals, d_count):
+    averages = {}
+    for item in donations:
+        averages[item] = d_totals[item]/d_count[item]
+    return averages
+
+def thanks():
+    while True:
+        name = input('Please enter the Full Name of the donor')
+        if name == 'list':
+            print(donations.keys())
+            continue
+        else:
+            break
+    donation = float(input('Please enter the amount of the donation'))
+    if name not in donations:
+        donations[name] = [donation]
+    else:
+        donations[name].append(donation)
+    print(f'Dear {name}:')
+    print(f'Thank you for your donation of ${donation:.2f} to our charity.')
+    print(f'Your contribution will do a great deal to help our worthy cause')
+
+def report():
+    title_str = '{:<20s}|{:^13}|{:^11}|{:>13}'
+    bar_str = '-'*60
+    entry_str = '{:<20s} ${:>12} {:>11} ${:>12}'
+    total_given = totals(donations)
+    num = count(donations)
+    avg = averages(total_given, num)
```

Comment:
this is ok for this assignment but normally you would want to consider performance, these are 3 additional loops that are not necessary.

you could have done sort on original dictionary and specify that you want to sort on the sum in the `sorted` function.

the avg, sum, total could be computed as part of the final loop for the report.